torch/csrc/autograd/variable.h

```diff
@@ -231,6 +238,12 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   std::vector<std::unique_ptr<FunctionPreHook>> hooks_;
   std::shared_ptr<hooks_list> cpp_hooks_list_;
 
+  // The post_acc_grad_hooks_ field stores only Python hooks that are
+  // called after the .grad field has been accumulated into. This is
+  // less complicated than the hooks_ field, which encapsulates
+  // a lot more.
+  std::vector<std::unique_ptr<PostAccumulateGradHook>> post_acc_grad_hooks_;
```

Comment:
This doesn't need to be a vector then? :D

Reply:
Somewhat of a preexisting issue lol

Reply:
Is it? The others can actually have multiple things at the same time (if you register python and c++ hooks)

Reply:
it is no longer a vector

Reply:
we actually currently clear the list before adding so that there's only one haha (totally silently wrong in the case some has both python and cpp though)
we can probably explicitly detect this by adding another field indicating whether there is already a cpp hook registered or python hook registered and error if you try to overwrite  hook of the other type.